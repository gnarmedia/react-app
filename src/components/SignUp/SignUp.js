/**
 * @file Sign Up component
 */

import React, { Component } from "react";

import Input from "../Input/Input";
import Button from "../Button/Button";

/**
 * Renders SignUp component.
 *
 * @example
 * <SignUp />
 *
 * @returns {Function} SignUp component.
 */
class SignUp extends Component {
  /**
   * Renders component.
   *
   * @returns {Function} - React component.
   */
  render() {
    return (
      <section className="is-fullheight">
        <div className="hero-body">
          <div className="container has-text-centered">
            <div className="column is-4 is-offset-4">
              <h3 className="title has-text-grey">Sign Up</h3>

              <div className="box">
                <form>
                  <Input name="email" placeholder="Your Email address" />
                  <Input
                    name="password"
                    placeholder="Your password"
                    type="password"
                  />
                  <Input
                    name="passwordConfirm"
                    placeholder="Confirm password"
                    type="password"
                  />
                  <Button>Sign Up</Button>
                </form>
              </div>

              <a href="/sign-in">Sign In</a>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default SignUp;
