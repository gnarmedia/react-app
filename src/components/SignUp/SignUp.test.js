/**
 * @file Sign Up component tests
 */

import React from 'react'
import { shallow } from 'enzyme'

import SignUp from './SignUp'

describe('SignUp', () => {
  it('should render the SignUp component', () => {
    const wrapper = shallow(<SignUp />)

    expect(wrapper).toMatchSnapshot()
  })
})
