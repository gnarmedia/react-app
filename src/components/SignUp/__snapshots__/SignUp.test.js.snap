// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render the SignUp component 1`] = `
<section
  className="is-fullheight"
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <div
        className="column is-4 is-offset-4"
      >
        <h3
          className="title has-text-grey"
        >
          Sign Up
        </h3>
        <div
          className="box"
        >
          <form>
            <Input
              name="email"
              placeholder="Your Email address"
              type="text"
            />
            <Input
              name="password"
              placeholder="Your password"
              type="password"
            />
            <Input
              name="passwordConfirm"
              placeholder="Confirm password"
              type="password"
            />
            <Button
              type="submit"
            >
              Sign Up
            </Button>
          </form>
        </div>
        <a
          href="/sign-in"
        >
          Sign In
        </a>
      </div>
    </div>
  </div>
</section>
`;
